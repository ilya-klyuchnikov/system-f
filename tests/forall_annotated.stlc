(@ (Lam (a) 5) int)

(@ (Lam (a) (lam ((x int) (y int)) (- x y))) int)

(Lam (a) (lam ((x a)) x))

(@ (Lam (a) (lam ((x a)) x)) int)
