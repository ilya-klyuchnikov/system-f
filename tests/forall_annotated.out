>  : int
-> 5
>  : (-> int int int)
-> #<function>
>  : (all (a) (-> a a))
-> #<function>
>  : (-> int int)
-> #<function>
> 